src/GraphQL/Client/Http.elm

```diff
@@ -51,182 +42,86 @@ type Error
-
-        options =
-            { method = "GET"
-            , headers = []
-            , url = "/graphql"
-            , timeout = Nothing
-            , withCredentials = False
-            }
+            Json.Decode.map2 (\errors data -> ( errors, data ))
+                (Json.Decode.maybe (Json.Decode.field "errors" Response.errorsDecoder))
+                (Json.Decode.field "data" (Builder.responseDataDecoder request))
     in
-        mutationRequest
-            |> GraphQL.Client.Http.customSendMutationRaw options
-            |> Task.andThen
-                (\response ->
-                    case Json.Decode.decodeString decoder response.body of
-                        Err err ->
-                            Task.fail <| GraphQL.Client.Http.HttpError <| Http.BadPayload err response
-
-                        Ok decodedValue ->
-                            Task.succeed decodedValue
-                )
-
--}
-customSendMutationRaw :
-    RequestOptions
-    -> Builder.Request Builder.Mutation result
-    -> Task Error (Http.Response String)
-customSendMutationRaw =
-    sendExpecting rawExpect
+    Http.expectStringResponse toMsg <|
+        \response ->
+            case response of
+                Http.BadUrl_ url ->
+                    Err (HttpError (Http.BadUrl url))
 
+                Http.Timeout_ ->
+                    Err (HttpError Http.Timeout)
 
-rawExpect : Http.Expect (Http.Response String)
-rawExpect =
-    Http.expectStringResponse Ok
+                Http.NetworkError_ ->
+                    Err (HttpError Http.NetworkError)
 
+                Http.BadStatus_ metadata body ->
+                    Err (HttpError (Http.BadStatus metadata.statusCode))
 
-send :
-    RequestOptions
-    -> Builder.Request operationType result
-    -> Task Error result
-send options request =
-    let
-        expect =
-            Util.defaultExpect (Builder.responseDataDecoder request)
-    in
-        sendExpecting expect options request
+                Http.GoodStatus_ metadata body ->
+                    case Json.Decode.decodeString decoder body of
+                        Ok ( Just errors, _ ) ->
+                            Err (GraphQLError errors)
 
+                        Ok ( Nothing, data ) ->
+                            Ok data
 
-sendExpecting :
-    Http.Expect result
-    -> RequestOptions
-    -> Builder.Request operationType result2
-    -> Task Error result
-sendExpecting expect requestOptions request =
-    let
-        documentString =
-            Builder.requestBody request
-
-        variableValues =
-            Builder.jsonVariableValues request
-    in
-        Util.requestConfig requestOptions documentString expect variableValues
-            |> Http.request
-            |> Http.toTask
-            |> Task.mapError (Util.convertHttpError HttpError GraphQLError)
+                        Err err ->
+                            Err (HttpError (Http.BadBody (Json.Decode.errorToString err)))
```

Comment:
It might make sense to also now flatten the `Error` type instead of nesting `Http.Error`, as its not really an /HTTP/ bad body.